pkgs/development/python-modules/ml-dtypes/default.nix

```diff
@@ -0,0 +1,33 @@
+{ buildPythonPackage
+, fetchFromGitHub
+, lib
+, numpy
+, pybind11
+}:
+
+buildPythonPackage rec {
+  pname = "ml-dtypes";
+  version = "0.2.0";
+  # As of 2023-06-06, they have both pyproject.toml and setup.py, seems like setuptools works more easily.
+  format = "setuptools";
+
+  src = fetchFromGitHub {
+    owner = "jax-ml";
+    repo = "ml_dtypes";
+    rev = "refs/tags/v${version}";
+    # ml-dtypes has eigen pinned as a submodule. My efforts to use nixpkgs's eigen have thus far failed :/
+    fetchSubmodules = true;
```

Comment:
Can we link our eigen source?

Reply:
I made an attempt at this but was not able to manage it. Happy to update if someone can figure out how to make it happen!